apps/site/assets/ts/schedule/components/line-diagram/LineDiagramWithStops.tsx

```diff
@@ -2,20 +2,27 @@ import React, { ReactElement } from "react";
 import { hasBranchLines } from "./line-diagram-helpers";
 import Diagram from "./graphics/Diagram";
 import StopListWithBranches from "./StopListWithBranches";
-import { CommonLineDiagramProps } from "./__line-diagram";
+import { LiveDataByStop } from "./__line-diagram";
+import { LineDiagramStop, RouteStop } from "../__schedule";
 import useStopPositions, { RefList } from "./graphics/useStopPositions";
 import StopCard from "./StopCard";
 import { hasPredictionTime } from "../../../models/prediction";
+import { MapMarker } from "../../../leaflet/components/__mapdata";
+
+interface Props {
+  stops: LineDiagramStop[];
+  handleStopClick: (stop: RouteStop) => void;
+  liveData: LiveDataByStop;
+  vehicleMarkers: MapMarker[];
```

Comment:
The way this component is set up right now, the entire line diagram component tree will re-render every time the channel updates any of the vehicle markers. That's a lot! :sweater